src/hal/drivers/mesa-hostmot2/hm2_rpspi.c

```diff
@@ -0,0 +1,568 @@
+	BCM2835_SPICS |= SPI_CS_TA|SPI_CS_CLEAR_TX|SPI_CS_CLEAR_RX;
+
+	/*rtapi_print_msg(RTAPI_MSG_ERR, "Out-Data: ");
+	for (i=0; i<msgsize; i++) {
+		rtapi_print_msg(RTAPI_MSG_ERR, "%08x:", this->txBuf[i]);
+	} 
+		rtapi_print_msg(RTAPI_MSG_ERR, "1 BCM2835_SPICS: %08x:", BCM2835_SPICS);
+	*/
+	
+	tbuff = (uint8_t *)this->txBuf;
+	rbuff = (uint8_t *)this->rxBuf;
+	
+	
+	/* send txBuf */
+	/* RPi SPI transfers 8 bits at a time only */
+	for (i=0; i<msgsize+1; i++) {
+		for(j=3; j>=0; j--){
+			//while(!(BCM2835_SPICS & SPI_CS_TXD));	// wait for  free space in fifo
+			BCM2835_SPIFIFO = tbuff[j];
+			if(!(BCM2835_SPICS & SPI_CS_TXD))
+				rtapi_print_msg(RTAPI_MSG_ERR, "TX-FIFO is full!\n");
+		}
+		
+		/* wait until transfer is finished */
+		while (!(BCM2835_SPICS & SPI_CS_DONE));
+		
+		//rtapi_print_msg(RTAPI_MSG_ERR, "2 BCM2835_SPICS: %08x:", BCM2835_SPICS);
+		
+		if((BCM2835_SPICS & SPI_CS_RXR) || (BCM2835_SPICS & SPI_CS_RXF))
+			rtapi_print_msg(RTAPI_MSG_ERR, "RX-FIFO is full!\n");
+		
+		for(j=3; j>=0; j--){
+			//while(!(BCM2835_SPICS & SPI_CS_RXD));	// wait for data in fifo
+			if(!(BCM2835_SPICS & SPI_CS_RXD))
+				rtapi_print_msg(RTAPI_MSG_ERR, "RX-FIFO is empty!\n");
+			rbuff[j] = BCM2835_SPIFIFO;
+		}
+		tbuff += 4;
+		rbuff += 4;
+	}
+	
+	/*rtapi_print_msg(RTAPI_MSG_ERR, "In-Data: ");
+	for (i=0; i<msgsize; i++) {
+		rtapi_print_msg(RTAPI_MSG_ERR, "%08x:", this->rxBuf[i]);
+	}
+	rtapi_print_msg(RTAPI_MSG_ERR, "\n"); */
+	
+	/* Stop transfer */
+	BCM2835_SPICS &= ~SPI_CS_TA;
+	
+	//rtapi_print_msg(RTAPI_MSG_ERR, "3 BCM2835_SPICS: %08x:", BCM2835_SPICS);
+	
+	/*buff = (char *)this->rxBuf;
+	buff += 4; //skip the cookie */
+	for (i=0; i<msgsize; i++) {
+		buffer32[i] = this->rxBuf[i+1];
+	}
+	
+    return 1;
+}
+
+/*************************************************/
+static int check_cookie(hm2_rpspi_t *board) {
+    uint32_t cookie[4];
+    uint32_t xcookie[4] = {0x55aacafe, 0x54534f48, 0x32544f4d, 0x00000400};
+    int r = hm2_rpspi_read(&board->llio, 0x100, cookie, 16);
+    if(r < 0) return -errno;
+
+    if(memcmp(cookie, xcookie, sizeof(cookie))) {
+        rtapi_print_msg(RTAPI_MSG_ERR, "Invalid cookie\n");
+        rtapi_print_msg(RTAPI_MSG_ERR, "Read: %08x %08x %08x %08x\n",
+            cookie[0], cookie[1], cookie[2], cookie[3]);
+        return -ENODEV;
+    }
+    return 0;
+}
+
+/*************************************************/
+static int read_ident(hm2_rpspi_t *board, char *ident) {
+    return hm2_rpspi_read(&board->llio, 0x40c, ident, 8);
+}
+
+/*************************************************/
+static int probe_board(hm2_rpspi_t *board, uint16_t spiclkdiv) {
+    //printf("probe %d\n", spiclkdiv);
+   
+    board->spiclkdiv = spiclkdiv;
+    board->spibpf = 32;
+
+    int ret = check_cookie(board);
+    if(ret < 0) return ret;
+
+    char ident[8];
+    ret = read_ident(board, ident);
+    if(ret < 0) return ret;
+
+    char *base;
+
+    if(!memcmp(ident, "MESA7I43", 8)) {
```

Comment:
I would prefer to just delete the code pertaining to 7i43 (from hm2_spi as well at this point).  

Background:  When I decided to do SPI support, I didn't have a 7i90 but I did have a 7i43, so (probably with a bit of help) I made a bitfile that did SPI on 7i43 for testing.  No bitfile ever shipped by mesa or by linuxcnc.org for the 7i43 supports SPI mode, I've lost my self-built bitfile that did, and because the SPI clock line requires a special FPGA pin (GCLK?) that is on one of the HD50 I/O connectors and not on the HD26/DB-25 connector it could never be cable-compatible with a 7i90 anyway, and leaves you with only one full standard I/O connector to use.